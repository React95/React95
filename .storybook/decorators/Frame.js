import React from 'react';

import { createGlobalStyle } from 'styled-components';

import woff2 from 'assets/font/MS-Sans-Serif.woff2';
import woff from 'assets/font/MS-Sans-Serif.woff';
import ttf from 'assets/font/MS-Sans-Serif.ttf';
import eot from 'assets/font/MS-Sans-Serif.eot';

import pattern from 'assets/pattern/dropdown.png';
import upcaret from 'assets/pattern/upcaret.svg';
import downcaret from 'assets/pattern/downcaret.svg';
import leftcaret from 'assets/pattern/leftcaret.svg';
import rightcaret from 'assets/pattern/rightcaret.svg';

import auto from 'assets/cursors/Arrow.png';
import text from 'assets/cursors/Text.png';
import verticalText from 'assets/cursors/VText.png';
import help from 'assets/cursors/Help.png';
import crosshair from 'assets/cursors/Cross.png';
import pointer from 'assets/cursors/HandPointer.png';
import progress from 'assets/cursors/AppStarting.png';
import wait from 'assets/cursors/Wait.png';
import alias from 'assets/cursors/Link.png';
import copy from 'assets/cursors/Copy.png';
import move from 'assets/cursors/Move.png';
import noDrop from 'assets/cursors/Circle.png';
import notAllowed from 'assets/cursors/NO.png';
import grab from 'assets/cursors/Hand.png';
import grabbing from 'assets/cursors/Handsqueezed.png';
import colResize from 'assets/cursors/ColRezise.png';
import rowResize from 'assets/cursors/RowResize.png';
import nResize from 'assets/cursors/UpArrow.png';
import eResize from 'assets/cursors/RightArrow.png';
import sResize from 'assets/cursors/DownArrow.png';
import wResize from 'assets/cursors/LeftArrow.png';
import nsResize from 'assets/cursors/UpDownArrow.png';
import ewResize from 'assets/cursors/LeftRight.png';
import neResize from 'assets/cursors/UpRightArrow.png';
import nwResize from 'assets/cursors/UpLeftArrow.png';
import seResize from 'assets/cursors/DownRightArrow.png';
import swResize from 'assets/cursors/DownLeftArrow.png';
import neswResize from 'assets/cursors/AngleUpRight.png';
import nwseResize from 'assets/cursors/AngleUpLeft.png';
import zoomIn from 'assets/cursors/ZoomIn.png';
import zoomOut from 'assets/cursors/ZoomOut.png';

const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'MS Sans Serif';
    src: url('${eot}');
    src: url('${woff2}') format('woff2'),
         url('${woff}') format('woff'),
         url('${ttf}') format('truetype'),
         url('${eot}.eot?#iefix') format('embedded-opentype');
    font-weight: normal;
    font-style: normal;
  }

  body {
    background-color: #5aa;
  }

  * {
    font-family: 'MS Sans Serif';
    font-size: 12px;
    box-sizing: border-box;
  }

  ::-webkit-scrollbar {
    width: 15px;
    height: 15px;
    background: url('${pattern}');
    background-size: 3px 3px;
  }

  ::-webkit-scrollbar-corner {
    background-color: transparent;
  }

  ::-webkit-scrollbar-button {
    width: 15px;
    height: 15px;
    background-color: #c3c7cb;
    box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e;
    border-right: 2px solid black;
    border-bottom: 2px solid black;

    background-size: 10px 10px;
    background-repeat: no-repeat;
    background-position: center center;
  }

  ::-webkit-scrollbar-button:end:decrement,
  ::-webkit-scrollbar-button:start:increment {
    display: none;
  }

  ::-webkit-scrollbar-button:vertical {
    background-size: 6px 3px;
  }

  ::-webkit-scrollbar-button:horizontal {
    background-size: 3px 6px;
  }

  ::-webkit-scrollbar-button:vertical:start:decrement {
    background-image: url('${upcaret}');
  }

  ::-webkit-scrollbar-button:vertical:end:increment {
    background-image: url('${downcaret}');
  }

  ::-webkit-scrollbar-button:horizontal:start:decrement {
    background-image: url('${leftcaret}');
  }

  ::-webkit-scrollbar-button:horizontal:end:increment {
    background-image: url('${rightcaret}');
  }

  ::-webkit-resizer {
    /* TODO? */
  }

  ::-webkit-scrollbar-thumb {
    background-color: #c3c7cb;
    box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e;
    border-right: 2px solid black;
    border-bottom: 2px solid black;
  }

  html, .auto, *   { cursor: url('${auto}'), auto; }
  .default         { cursor: url('${auto}'), default; }
  .none            { cursor: none; }
  .help            { cursor: url('${help}'), help; }
  .pointer         { cursor: url('${pointer}'), pointer; }
  .progress        { cursor: url('${progress}'), progress; }
  .wait            { cursor: url('${wait}'), wait; }
  .crosshair       { cursor: url('${crosshair}'), crosshair; }
  .text            { cursor: url('${text}'), text; }
  .vertical-text   { cursor: url('${verticalText}'), vertical-text; }
  .alias           { cursor: url('${alias}'), alias; }
  .copy            { cursor: url('${copy}'), copy; }
  .move            { cursor: url('${move}'), move; }
  .no-drop         { cursor: url('${noDrop}'), no-drop; }
  .not-allowed     { cursor: url('${notAllowed}'), not-allowed; }
  .grab            { cursor: url('${grab}'), grab; }
  .grabbing        { cursor: url('${grabbing}'), grabbing; }
  .col-resize      { cursor: url('${colResize}'), col-resize; }
  .row-resize      { cursor: url('${rowResize}'), row-resize; }
  .n-resize        { cursor: url('${nResize}'), n-resize; }
  .e-resize        { cursor: url('${eResize}'), e-resize; }
  .s-resize        { cursor: url('${sResize}'), s-resize; }
  .w-resize        { cursor: url('${wResize}'), w-resize; }
  .ns-resize       { cursor: url('${nsResize}'), ns-resize; }
  .ew-resize       { cursor: url('${ewResize}'), ew-resize; }
  .ne-resize       { cursor: url('${neResize}'), ne-resize; }
  .nw-resize       { cursor: url('${nwResize}'), nw-resize; }
  .se-resize       { cursor: url('${seResize}'), se-resize; }
  .sw-resize       { cursor: url('${swResize}'), sw-resize; }
  .nesw-resize     { cursor: url('${neswResize}'), nesw-resize; }
  .nwse-resize     { cursor: url('${nwseResize}'), nwse-resize; }
  .zoom-in         { cursor: url('${zoomIn}'), zoom-in; }
  .zoom-out        { cursor: url('${zoomOut}'), zoom-out; }
`;

const Frame = storyFn => (
  <React.Fragment>
    <GlobalStyle />
    {storyFn()}
  </React.Fragment>
);

export default Frame;
