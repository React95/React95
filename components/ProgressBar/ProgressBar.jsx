import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

const Wrapper = styled.div`
  width: ${props => props.width}px;
  height: 20px;

  position: relative;

  text-align: center;
`;

const WhiteBar = styled.div`
  width: ${props => props.width}px;
  height: 20px;
  line-height: 20px;

  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;

  background: #fff;
  color: #000;

  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000,
    0.5px 0.5px 0 0.5px #ffffff;
`;

const Container = styled.div`
  width: ${props => props.percent}%;

  position: absolute;
  top: 0;
  left: 0;

  overflow: hidden;
`;

const BlueBar = styled.div`
  width: ${props => props.width}px;
  height: 17px;
  line-height: 18px;

  margin-left: 2px;
  margin-top: 2px;

  background: #000e7a;
  color: #fff;
`;

const ProgressBar = ({ width, percent }) => (
  <Wrapper width={width}>
    <WhiteBar width={width}>
      {percent}
%
    </WhiteBar>
    <Container percent={percent}>
      <BlueBar width={width}>
        {percent}
%
      </BlueBar>
    </Container>
  </Wrapper>
);

ProgressBar.propTypes = {
  width: PropTypes.number,
  percent: PropTypes.number,
};

ProgressBar.defaultProps = {
  width: 150,
  percent: 0,
};

export default ProgressBar;
