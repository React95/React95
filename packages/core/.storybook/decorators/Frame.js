import React from 'react';
import { GlobalStyle } from '../../components';

const Frame = Story => (
  <div style={{ padding: 10 }}>
    <GlobalStyle />
    <Story />
  </div>
);

export default Frame;
