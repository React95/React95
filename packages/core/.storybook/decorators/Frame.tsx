import React from 'react';
import '../../components/GlobalStyle/GlobalStyle.css';
import * as themes from '../../components/themes/all';

const Frame = (Story, { globals }) => {
  const { selectedTheme } = globals;

  const theme = themes[selectedTheme] || themes.win95;

  React.useEffect(() => {
    document.body.classList.add(theme);

    return () => {
      document.body.classList.remove(theme);
    };
  }, [theme]);

  return (
    <div style={{ padding: 10 }}>
      <Story />
    </div>
  );
};

export default Frame;
