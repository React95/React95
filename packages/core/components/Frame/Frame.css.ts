import { createRainbowSprinkles, defineProperties } from 'rainbow-sprinkles';
import {
  background,
  borderRadius,
  borders,
  colors,
  displayAndBoxModel,
  font,
  outline,
  positioning,
  shadows,
  text,
  zIndices,
} from './props';
import type { SprinklesFn } from 'rainbow-sprinkles/createRuntimeFn';

const displayAndBoxModelAtomicProps = defineProperties({
  dynamicProperties: displayAndBoxModel,
  shorthands: {
    size: ['height', 'width'],
    h: ['height'],
    w: ['width'],
    minH: ['minHeight'],
    minW: ['minWidth'],

    m: ['margin'],
    mr: ['marginRight'],
    ml: ['marginLeft'],
    mt: ['marginTop'],
    mb: ['marginBottom'],
    marginX: ['marginLeft', 'marginRight'],
    marginY: ['marginTop', 'marginBottom'],
    mx: ['marginLeft', 'marginRight'],
    my: ['marginTop', 'marginBottom'],

    p: ['padding'],
    pr: ['paddingRight'],
    pl: ['paddingLeft'],
    pt: ['paddingTop'],
    pb: ['paddingBottom'],
    paddingX: ['paddingLeft', 'paddingRight'],
    paddingY: ['paddingTop', 'paddingBottom'],
    px: ['paddingLeft', 'paddingRight'],
    py: ['paddingTop', 'paddingBottom'],
  },
});

const positioningAtomicProps = defineProperties({
  dynamicProperties: positioning,
});

const colorsAtomicProps = defineProperties({
  dynamicProperties: colors,
  shorthands: {
    bgColor: ['backgroundColor'],
  },
});

const backgroundAtomicProps = defineProperties({
  dynamicProperties: background,
  shorthands: {
    bg: ['background'],
  },
});

const bordersAtomicProps = defineProperties({
  dynamicProperties: borders,
});

const borderRadiusAtomicProps = defineProperties({
  dynamicProperties: borderRadius,
});

const shadowsAtomicProps = defineProperties({
  dynamicProperties: shadows,
});

const outlineAtomicProps = defineProperties({
  dynamicProperties: outline,
});

const fontAtomicProps = defineProperties({
  dynamicProperties: font,
});

const textAtomicProps = defineProperties({
  dynamicProperties: text,
});

const zIndicesAtomicProps = defineProperties({
  dynamicProperties: zIndices,
});

export const sprinkles: SprinklesFn<
  [
    typeof colorsAtomicProps,
    typeof shadowsAtomicProps,
    typeof zIndicesAtomicProps,
    typeof displayAndBoxModelAtomicProps,
    typeof positioningAtomicProps,
    typeof backgroundAtomicProps,
    typeof bordersAtomicProps,
    typeof borderRadiusAtomicProps,
    typeof outlineAtomicProps,
    typeof fontAtomicProps,
    typeof textAtomicProps,
  ]
> = createRainbowSprinkles(
  colorsAtomicProps,
  shadowsAtomicProps,
  zIndicesAtomicProps,
  displayAndBoxModelAtomicProps,
  positioningAtomicProps,
  backgroundAtomicProps,
  bordersAtomicProps,
  borderRadiusAtomicProps,
  outlineAtomicProps,
  fontAtomicProps,
  textAtomicProps,
);

export type Sprinkles = Parameters<typeof sprinkles>[0];
