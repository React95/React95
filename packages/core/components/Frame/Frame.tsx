import React, {
  ComponentPropsWithRef,
  ElementType,
  ForwardedRef,
  forwardRef,
} from 'react';
import { Sprinkles, sprinkles } from './sprinkles.css';
import cn from 'classnames';

type FixedForwardRef = <T, P = object>(
  render: (props: P, ref: React.Ref<T>) => React.ReactNode,
) => (props: P & React.RefAttributes<T>) => React.ReactNode;

const fixedForwardRef = forwardRef as FixedForwardRef;

type DistributiveOmit<T, TOmitted extends PropertyKey> = T extends any
  ? Omit<T, TOmitted>
  : never;

export type FrameProps<TAs extends ElementType = 'div'> = {
  as?: TAs;
} & DistributiveOmit<
  ComponentPropsWithRef<ElementType extends TAs ? 'div' : TAs>,
  'as'
> &
  Sprinkles;

const FrameComponent = <TAs extends ElementType>(
  props: FrameProps<TAs>,
  ref: ForwardedRef<any>,
) => {
  const { as, children, ...rest } = props;
  const Component = as || 'div';
  const { className, style, otherProps } = sprinkles(rest);

  return (
    <Component
      {...otherProps}
      style={{ ...style, ...otherProps.style }}
      className={cn(className, otherProps.className)}
      ref={ref}
    >
      {children}
    </Component>
  );
};

export const Frame = fixedForwardRef(FrameComponent);
