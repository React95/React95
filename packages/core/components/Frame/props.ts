import { contract } from '../themes/contract.css';

export const positioning = {
  position: true,
  top: true,
  right: true,
  bottom: true,
  left: true,
  pointerEvents: true,
} as const;

export const displayAndBoxModel = {
  display: true,
  overflow: true,
  boxSizing: true,

  grid: true,
  gridArea: true,
  gridAutoColumns: true,
  gridAutoFlow: true,
  gridAutoRows: true,
  gridColumn: true,
  gridColumnEnd: true,
  gridColumnStart: true,
  gridRow: true,
  gridRowEnd: true,
  gridRowStart: true,
  gridTemplate: true,
  gridTemplateAreas: true,
  gridTemplateColumns: true,
  gridTemplateRows: true,

  flex: true,
  flexBasis: true,
  flexDirection: true,
  flexFlow: true,
  flexGrow: true,
  flexShrink: true,
  flexWrap: true,

  justifyContent: true,
  justifyItems: true,
  justifySelf: true,
  alignContent: true,
  alignItems: true,
  alignSelf: true,

  gap: contract.space,

  width: contract.space,
  maxWidth: contract.space,
  minWidth: contract.space,
  height: contract.space,
  maxHeight: contract.space,
  minHeight: contract.space,

  margin: contract.space,
  marginBlock: contract.space,
  marginBlockEnd: contract.space,
  marginBlockStart: contract.space,
  marginBottom: contract.space,
  marginInline: contract.space,
  marginInlineEnd: contract.space,
  marginInlineStart: contract.space,
  marginLeft: contract.space,
  marginRight: contract.space,
  marginTop: contract.space,

  padding: contract.space,
  paddingBlock: contract.space,
  paddingBlockEnd: contract.space,
  paddingBlockStart: contract.space,
  paddingBottom: contract.space,
  paddingInline: contract.space,
  paddingInlineEnd: contract.space,
  paddingInlineStart: contract.space,
  paddingLeft: contract.space,
  paddingRight: contract.space,
  paddingTop: contract.space,
} as const;

export const colors = {
  backgroundColor: contract.colors,
  color: contract.colors,
} as const;

export const background = {
  background: true,
  backgroundAttachment: true,
  backgroundImage: true,
  backgroundPosition: true,
  backgroundRepeat: true,
  backgroundSize: true,
  backgroundOrigin: true,
  backgroundClip: true,
} as const;

export const borders = {
  border: true,
  borderStyle: true,
  borderWidth: true,
  borderColor: contract.colors,
  borderTop: true,
  borderTopColor: contract.colors,
  borderTopStyle: true,
  borderTopWidth: true,
  borderRight: true,
  borderRightColor: contract.colors,
  borderRightStyle: true,
  borderRightWidth: true,
  borderBottom: true,
  borderBottomColor: contract.colors,
  borderBottomStyle: true,
  borderBottomWidth: true,
  borderLeft: true,
  borderLeftColor: contract.colors,
  borderLeftStyle: true,
  borderLeftWidth: true,
} as const;

export const borderRadius = {
  borderBottomLeftRadius: true,
  borderBottomRightRadius: true,
  borderImage: true,
  borderImageOutset: true,
  borderImageRepeat: true,
  borderImageSlice: true,
  borderImageSource: true,
  borderImageWidth: true,
  borderRadius: true,
  borderTopLeftRadius: true,
  borderTopRightRadius: true,
} as const;

export const shadows = {
  boxShadow: contract.shadows,
} as const;

export const outline = {
  outline: true,
  outlineColor: true,
  outlineStyle: true,
  outlineWidth: true,
} as const;

export const font = {
  font: true,
  fontFamily: true,
  fontStyle: true,
  fontSize: true,
  fontWeight: true,
  fontVariant: true,
  fontSizeAdjust: true,
} as const;

export const text = {
  color: contract.colors,
  direction: true,
  letterSpacing: true,
  lineHeight: true,
  textAlign: true,
  textDecoration: true,
  textIndent: true,
  textShadow: true,
  textTransform: true,
  verticalAlign: true,
  whiteSpace: true,
  wordSpacing: true,
  wordBreak: true,
} as const;

export const zIndices = {
  zIndex: contract.zIndices,
} as const;
