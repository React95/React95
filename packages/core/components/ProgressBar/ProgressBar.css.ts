import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';

export const wrapper = style({
  height: contract.space[20],
  position: 'relative',
  textAlign: 'center',
});

export const whiteBar = style({
  height: contract.space[20],
  lineHeight: contract.space[20],
  borderLeft: contract.space[1],
  borderLeftStyle: 'solid',
  borderLeftColor: contract.colors.borderDark,
  borderTop: contract.space[1],
  borderTopStyle: 'solid',
  borderTopColor: contract.colors.borderDark,
  backgroundColor: contract.colors.inputBackground,
  color: contract.colors.materialText,
  boxShadow: contract.shadows.input,
});

export const container = style({
  position: 'absolute',
  top: '0',
  left: '0',
  overflow: 'hidden',
});

export const progress = style({
  height: contract.space[17],
  lineHeight: contract.space[18],
  marginLeft: contract.space[2],
  marginTop: contract.space[2],
  backgroundColor: contract.colors.progress,
  color: contract.colors.materialTextInvert,
});
