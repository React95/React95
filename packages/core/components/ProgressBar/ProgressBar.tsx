import React, { forwardRef } from 'react';
import type {
  ElementType,
  ReactElement,
  ElementRef,
  ForwardedRef,
} from 'react';
import cn from 'classnames';

import { Frame, FrameProps } from '../Frame/Frame';
import * as styles from './ProgressBar.css';

export type ProgressBarProps<TAs extends ElementType> = {
  percent?: number;
} & FrameProps<TAs>;

export const ProgressBar = forwardRef<HTMLDivElement, ProgressBarProps<'div'>>(
  ({ width = '150px', percent = 0, ...rest }, ref) => (
    <Frame
      {...rest}
      width={width}
      className={cn(styles.wrapper, rest.className)}
      ref={ref}
    >
      <Frame className={styles.whiteBar} width={width}>{`${percent}%`}</Frame>
      <Frame className={styles.container} width={`${percent}%`}>
        <Frame className={styles.progress} width={width}>{`${percent}%`}</Frame>
      </Frame>
    </Frame>
  ),
) as <TAs extends ElementType = 'div'>(
  props: ProgressBarProps<TAs> & { ref?: ForwardedRef<ElementRef<TAs>> },
) => ReactElement;
