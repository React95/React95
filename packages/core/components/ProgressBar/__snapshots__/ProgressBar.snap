// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProgressBar /> > Snapshots > should match snapshot 1`] = `
<div>
  <div
    class="sprinkles_width__1txm3g81t ProgressBar_wrapper__1drc6y70"
    style="--width__1txm3g81s: 300px;"
  >
    <div
      class="sprinkles_width__1txm3g81t ProgressBar_whiteBar__1drc6y71"
      style="--width__1txm3g81s: 300px;"
    >
      50%
    </div>
    <div
      class="sprinkles_width__1txm3g81t ProgressBar_container__1drc6y72"
      style="--width__1txm3g81s: 50%;"
    >
      <div
        class="sprinkles_width__1txm3g81t ProgressBar_progress__1drc6y73"
        style="--width__1txm3g81s: 300px;"
      >
        50%
      </div>
    </div>
  </div>
</div>
`;
